src/goto-instrument/cover_basic_blocks.cpp

```diff
@@ -228,3 +160,46 @@ void cover_basic_blockst::update_covered_lines(block_infot &block_info)
   std::string covered_lines = format_number_range(line_list);
   block_info.source_location.set_basic_block_covered_lines(covered_lines);
 }
+
+cover_basic_blocks_javat::cover_basic_blocks_javat(
+  const goto_programt &_goto_program)
+{
+  forall_goto_program_instructions(it, _goto_program)
+  {
+    const auto &bytecode_index = it->source_location.get_java_bytecode_index();
+    if(!index_to_block.count(bytecode_index))
+    {
+      block_infos.push_back(it);
+      index_to_block.emplace(bytecode_index, block_infos.size() - 1);
```

Comment:
That's two lookups in quick succession. Instead:
````c++
auto &entry = index_to_block.insert({bytecode_index, block_infos.size() - 1});
if(entry.second)
  block_infos.push_back(it);
````

Reply:
Well spotted (except size-1 needs to be changed to size)